src/state/mouseEventsMiddleware.ts

```diff
@@ -34,18 +34,25 @@ mouseMoveMiddleware.startListening({
   actionCreator: mouseMoveAction,
   effect: (action: PayloadAction<MousePointer>, listenerApi: ListenerEffectAPI<RechartsRootState, AppDispatch>) => {
     const mousePointer = action.payload;
-    const activeProps = selectActivePropsFromMousePointer(listenerApi.getState(), mousePointer);
-    if (activeProps?.activeIndex != null) {
-      listenerApi.dispatch(
-        setMouseOverAxisIndex({
-          activeIndex: activeProps.activeIndex,
-          activeDataKey: undefined,
-          activeMouseOverCoordinate: activeProps.activeCoordinate,
-        }),
-      );
-    } else {
-      // this is needed to clear tooltip state when the mouse moves out of the inRange (svg - offset) function, but not yet out of the svg
-      listenerApi.dispatch(mouseLeaveChart());
+    const state = listenerApi.getState();
+    // can we even check for the 'shared' prop here? This will be wrong if it is used... Can't use hooks.
```

Comment:
Can't use hooks here but you can refactor the hook to be a selector and then call that selector: https://github.com/recharts/recharts/blob/3.x/src/state/selectors/selectors.ts#L61

Reply:
And yes if we need to access `shared` from outside of the Tooltip element itself, then we're going to need `TooltipSettingsState` or something and read it from there. I try to avoid putting stuff into the global redux store unless necessary and well ... it wasn't necessary until now I guess.

Reply:
Got it, will fix